File: phase1-tests/rust/test_program.rs
use std::collections::HashMap;

fn main() {
    let mut passed = 0;
    let mut failed = 0;

    // Test 1: Basic output
    println!("Hello from Hummingbird Rust!");
    passed += 1;
    println!("  Basic Output - PASSED");

    // Test 2: JSON-like data structure
    let mut data: HashMap<&str, &str> = HashMap::new();
    data.insert("name", "test");
    data.insert("value", "123");
    if data.get("name") == Some(&"test") && data.get("value") == Some(&"123") {
        println!("  HashMap Operations - PASSED");
        passed += 1;
    } else {
        println!("  HashMap Operations - FAILED");
        failed += 1;
    }

    // Test 3: String processing
    let text = "hummingbird rust test";
    let upper = text.to_uppercase();
    if upper == "HUMMINGBIRD RUST TEST" {
        println!("  String Processing - PASSED");
        passed += 1;
    } else {
        println!("  String Processing - FAILED: got {}", upper);
        failed += 1;
    }

    // Test 4: File operations
    let tmp_path = "/tmp/rust-test-file.txt";
    match std::fs::write(tmp_path, "test content\n") {
        Ok(_) => {
            match std::fs::read_to_string(tmp_path) {
                Ok(content) if content == "test content\n" => {
                    println!("  File Operations - PASSED");
                    passed += 1;
                }
                _ => {
                    println!("  File Operations - FAILED: read mismatch");
                    failed += 1;
                }
            }
            let _ = std::fs::remove_file(tmp_path);
        }
        Err(e) => {
            println!("  File Operations - FAILED: {}", e);
            failed += 1;
        }
    }

    println!("\nResults: {} passed, {} failed", passed, failed);
    if failed > 0 {
        std::process::exit(1);
    }
}

File: phase1-tests/openjdk-25/TestProgram.java
import java.util.HashMap;
import java.io.File;
import java.io.FileWriter;
import java.io.BufferedReader;
import java.io.FileReader;

public class TestProgram {
    static int passed = 0;
    static int failed = 0;

    public static void main(String[] args) {
        // Test 1: Basic output
        System.out.println("Hello from Hummingbird OpenJDK 25!");
        passed++;
        System.out.println("  Basic Output - PASSED");

        // Test 2: HashMap operations
        try {
            HashMap<String, Object> data = new HashMap<>();
            data.put("name", "test");
            data.put("value", 123);
            if ("test".equals(data.get("name")) && Integer.valueOf(123).equals(data.get("value"))) {
                System.out.println("  HashMap Operations - PASSED");
                passed++;
            } else {
                System.out.println("  HashMap Operations - FAILED");
                failed++;
            }
        } catch (Exception e) {
            System.out.println("  HashMap Operations - FAILED: " + e.getMessage());
            failed++;
        }

        // Test 3: String processing
        String text = "hummingbird java test";
        if (text.toUpperCase().equals("HUMMINGBIRD JAVA TEST")) {
            System.out.println("  String Processing - PASSED");
            passed++;
        } else {
            System.out.println("  String Processing - FAILED");
            failed++;
        }

        // Test 4: File operations
        try {
            File tmpFile = new File("/tmp/java-test-file.txt");
            FileWriter writer = new FileWriter(tmpFile);
            writer.write("test content\n");
            writer.close();

            BufferedReader reader = new BufferedReader(new FileReader(tmpFile));
            String content = reader.readLine();
            reader.close();
            tmpFile.delete();

            if ("test content".equals(content)) {
                System.out.println("  File Operations - PASSED");
                passed++;
            } else {
                System.out.println("  File Operations - FAILED: got " + content);
                failed++;
            }
        } catch (Exception e) {
            System.out.println("  File Operations - FAILED: " + e.getMessage());
            failed++;
        }

        System.out.println("\nResults: " + passed + " passed, " + failed + " failed");
        if (failed > 0) System.exit(1);
    }
}
